integration-tests/test_ftp.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import unicode_literals
+
+from smart_open import open
+
+def test_nonbinary():
+    file_contents = "Test Test \n new test \n another tests"
```

Comment:
Could we add some tests for CLRF and B_CLRF? I noticed there is quite a lot of explicit handling of these in ftplib
https://github.com/python/cpython/blob/f00645d5dbf4cfa0b8f382c8977724578dff191d/Lib/ftplib.py#L70

Reply:
That is a good point. I have added the tests `test_line_endings_non_binary` and `test_line_endings_binary`. My logic for these tests is that, if you look at retrlines vs retrbinary in ftplib [here](https://github.com/python/cpython/blob/main/Lib/ftplib.py#L421), you will notice that `retrlines` explicitly removes CRLF line endings for each line, but `retrbinary` does not. As a result, in both of the aforementioned tests, I test to make sure that when the file is read in non binary mode, there is no CRLF ending in any line, but when the file is read in binary mode, each line contains the CRLF endings (in order to make sure that the behavior mimics `ftplib`). 

I have tested ftplib `retrlines` vs `retrbinary` and have confirmed that, if you upload a file with CRLF endings, then calling `retrbinary` on that file will result in each line containing a CRLF ending, but when calling `retrlines` each line will not contain those endings.